contracts/src/v0.1/BatchSubmission.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IAggregatorSubmit.sol";
+
+contract BatchSubmission is Ownable {
+    uint256 maxSubmission = 50;
+    mapping(address => bool) public oracleAddresses;
+
+    error OnlyOracle();
+    error InvalidOracle();
+    error InvalidLength();
+
+    event OracleAdded(address oracle);
+    event OracleRemoved(address oracle);
+    event MaxSubmissionSet(uint256 maxSubmission);
+    event Submited(uint256 aggreagtorAmount);
+
+    constructor() {}
+
+    function addOracle(address _oracle) public onlyOwner {
```

Comment:
I think it would be good to keep track of all oracles. How about keeping their addresses in a list?